pylint/config/_breaking_changes.py

```diff
@@ -0,0 +1,95 @@
+
+"""List the breaking changes in configuration files and their solutions."""
+
+from __future__ import annotations
+
+import enum
+from typing import NamedTuple
+
+
+class BreakingChange(enum.Enum):
+    MESSAGE_MADE_DISABLED_BY_DEFAULT = "{symbol} ({msgid}) was disabled by default"
+    MESSAGE_MADE_ENABLED_BY_DEFAULT = "{symbol} ({msgid}) was enabled by default"
+    MESSAGE_MOVED_TO_EXTENSION = "{symbol} ({msgid}) was moved to {extension}"
+    EXTENSION_REMOVED = "{extension} was removed"
+    # This kind of upgrade is non-breaking but if we want to automatically upgrade it,
+    # then we should use the message store and old_names values instead of duplicating
+    # MESSAGE_RENAMED= "{symbol} ({msgid}) was renamed"
+
+
+class Condition(enum.Enum):
+    MESSAGE_IS_ENABLED = "{symbol} ({msgid}) is enabled"
+    MESSAGE_IS_NOT_ENABLED = "{symbol} ({msgid}) is not enabled"
+    MESSAGE_IS_DISABLED = "{symbol} ({msgid}) is disabled"
+    MESSAGE_IS_NOT_DISABLED = "{symbol} ({msgid}) is not disabled"
+    EXTENSION_IS_LOADED = "{extension} is loaded"
+    EXTENSION_IS_NOT_LOADED = "{extension} is not loaded"
+
+
+class Information(NamedTuple):
+    symbol: str
+    msgid: str
+    extension: str | None
+
+
+class Solution(enum.Enum):
+    ADD_EXTENSION = "Add {extension} in 'load-plugins' option"
+    REMOVE_EXTENSION = "Remove {extension} from the 'load-plugins' option"
+    ENABLE_MESSAGE_EXPLICITLY = (
+        "{symbol} ({msgid}) should be added in the 'enable' option"
+    )
+    ENABLE_MESSAGE_IMPLICITLY = (
+        "{symbol} ({msgid}) should be removed from the 'disable' option"
+    )
+    DISABLE_MESSAGE_EXPLICITLY = (
+        "{symbol} ({msgid}) should be added in the 'disable' option"
+    )
+    DISABLE_MESSAGE_IMPLICITLY = (
+        "{symbol} ({msgid}) should be removed from the 'enable' option"
+    )
+
+
+ConditionsToBeAffected = list[Condition]
+# A solution to a breaking change might imply multiple actions
+MultipleActionSolution = list[Solution]
+# Sometimes there's multiple solutions and the user need to choose
+Solutions = list[MultipleActionSolution]
+BreakingChangeWithSolution = tuple[
+    BreakingChange, Information, ConditionsToBeAffected, Solutions
+]
+
+NO_SELF_USE = Information(
+    msgid="R6301", symbol="no-self-use", extension="pylint.extensions.no_self_use"
+)
+COMPARE_TO_ZERO = Information(
+    msgid="C2001", symbol="compare-to-zero", extension="pylint.extensions.comparetozero"
+)
+COMPARE_TO_EMPTY_STRING = Information(
+    msgid="C1901",
+    symbol="compare-to-empty-string",
+    extension="pylint.extensions.emptystring",
+)
+
+CONFIGURATION_BREAKING_CHANGES: list[BreakingChangeWithSolution] = [
+    (
+        BreakingChange.MESSAGE_MOVED_TO_EXTENSION,
+        NO_SELF_USE,
+        [Condition.MESSAGE_IS_NOT_DISABLED, Condition.EXTENSION_IS_NOT_LOADED],
+        [[Solution.ADD_EXTENSION], [Solution.DISABLE_MESSAGE_IMPLICITLY]],
```

Comment:
```suggestion
        [Condition.MESSAGE_IS_NOT_DISABLED, Condition.EXTENSION_IS_NOT_LOADED],
        [[Solution.ADD_EXTENSION], [Solution.DISABLE_MESSAGE_EXPLICITLY]],
```
Otherwise it's hard to know if the message was taken into account. We might want another way to tell if the pylintrc need upgrading, like maybe a ``upgrade-to=3.0.0`` option and keeping track of the version in the data structure.